### ISSUE TITLE
DependencyTracker sees tasks with no runs

### ISSUE THREAD

**Issue by djmitche** [taskcluster/taskcluster#3337]
https://sentry.prod.mozaws.net/operations/taskcluster-cloudops-stage/issues/9317900/
```
    // Put message in appropriate azure queue, and publish message to pulse,
    // if the initial run is pending
    if (task.runs[0].state === 'pending') { // <----
      await Promise.all([
        this.queueService.putPendingMessage(task, 0),
```
```
Cannot read property 'state' of undefined
```

**Comment by djmitche** [comment]
This has only happened once so far.  No record of which task it was.  But not (yet) an emergency.  This was some time after ugprading to v35.

**Comment by djmitche** [comment]
Guessing from the timestamps (error was at 16:32:34 UTC), this involved [HpVUYVvYR0SU_CsGw2W8jw](https://stage.taskcluster.nonprod.cloudops.mozgcp.net/tasks/HpVUYVvYR0SU_CsGw2W8jw) which started at 16:32:34 and depends on several tasks, of which [PdAroC-KRXCAl1MeH5boBw](https://stage.taskcluster.nonprod.cloudops.mozgcp.net/tasks/PdAroC-KRXCAl1MeH5boBw) finished at 16:32:30.

**Comment by djmitche** [comment]
Digging into the code here: `DependencyTracker.scheduleTask` is called to add the first run to an as-yet unscheduled task.  Which makes sense here: PdAroC-KRXCAl1MeH5boBw didn't have any runs until its dependencies all completed at 16:32:34.
https://github.com/taskcluster/taskcluster/blob/48080995d2663ec2c3b9c4776d3b2b58a3b1d60d/services/queue/src/dependencytracker.js#L253-L262
So that calls `db.fns.schedule_task` and passes the result to `task.updateStatusWith`:
https://github.com/taskcluster/taskcluster/blob/48080995d2663ec2c3b9c4776d3b2b58a3b1d60d/services/queue/src/data.js#L172-L179
The idea here is that the DB method returns an updated version of the task on success, and `updateStatusWith` updates the corresponding JS fields.  That "on success" is probably critical -- if `db.fns.schedule_task` was not successful, then it would have returned an empty set, and `updateStatusWith` would not have changed `task`, leaving `runs` an empty array and causing this exception.  So what would cause `db.fns.schedule_task` to return an empty set?
https://github.com/taskcluster/taskcluster/blob/48080995d2663ec2c3b9c4776d3b2b58a3b1d60d/db/versions/0028.yml#L360-L395
Either the task row wasn't found or there was already a run in place.  The former is unlikely -- we *just* loaded this task from the DB earlier in `scheduleTask`.

Looking at the logging:
```
2020-08-04 16:32:34.415 GMT db-function-call get_dependent_tasks taskcluster-queue-dependencyresolver 
2020-08-04 16:32:34.417 GMT db-function-call get_dependent_tasks taskcluster-queue-dependencyresolver 
2020-08-04 16:32:34.436 GMT db-function-call satisfy_task_dependency taskcluster-queue-dependencyresolver 
2020-08-04 16:32:34.437 GMT db-function-call satisfy_task_dependency taskcluster-queue-dependencyresolver 
2020-08-04 16:32:34.440 GMT db-function-call is_task_blocked taskcluster-queue-dependencyresolver 
2020-08-04 16:32:34.441 GMT db-function-call is_task_blocked taskcluster-queue-dependencyresolver 
2020-08-04 16:32:34.443 GMT db-function-call get_task taskcluster-queue-dependencyresolver 
2020-08-04 16:32:34.443 GMT db-function-call get_task taskcluster-queue-dependencyresolver 
2020-08-04 16:32:34.448 GMT db-function-call schedule_task taskcluster-queue-dependencyresolver 
2020-08-04 16:32:34.448 GMT db-function-call schedule_task taskcluster-queue-dependencyresolver 
2020-08-04 16:32:34.451 GMT db-function-call azure_queue_put taskcluster-queue-dependencyresolver                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           
2020-08-04 16:32:34.460 GMT monitor.error TypeError taskcluster-queue-dependencyresolver "TypeError: Cannot read property 'state' of undefined at ..
```
it sure looks like it's acting on the same message twice at the same time!

So that suggests either two identical task-resolved messages were pushed to that azqueue, or the `azure_queue_get` returned the same row to multiple callers (an error in the SQL queue implementation in tc-lib-azqueue).  We do call four `azure_queue_get` in parallel, probably as a leftover from Azure.

Looking at logging from 16:32:30, I only see one `reportCompleted` API call for this task, and only one corresponding `azure_queue_put`.  So it seems that `azure_queue_get` is improperly locking rows.

At any rate, we should be resilient to this, as pushing the same message multiple times is allowed as part of the idempotency of the `reportCompleted` method.  So I'll fix that up here, and filed a new issue to track the possible bug in `azure_queue_get`.
